validation/src/shared_table/mod.rs

```diff
@@ -306,7 +307,25 @@ impl<Fetch: Future> ParachainWork<Fetch> {
 			);
 
 			match res {
-				Ok(e) => Ok(e),
+				Ok((messages, fees)) => {
+					match crate::collation::validate_receipt(
+						&*api,
+						id,
+						&relay_parent,
+						collation,
+						receipt,
+						&messages,
+						fees,
+					) {
+						Ok(chunks) => {
+							Ok((messages, chunks.into_iter().nth(local_index).unwrap()))
```

Comment:
`O(n)` access rather than indexing? and the unwrap should be removed in favor of a provable `expect`